src/python/pants/backend/go/module.py

```diff
@@ -256,15 +260,65 @@ async def download_external_module(
             PathGlobs(
                 [f"{source_path}/**"],
                 glob_match_error_behavior=GlobMatchErrorBehavior.error,
-                description_of_origin=f"the DownloadExternalModuleRequest for {request.path}@{request.version}{request.path}@{request.version}",
+                description_of_origin=f"the DownloadExternalModuleRequest for {request.path}@{request.version}",
             ),
         ),
     )
 
     source_digest_stripped = await Get(Digest, RemovePrefix(source_digest, source_path))
 
+    source_snapshot = await Get(Snapshot, Digest, source_digest_stripped)
```

Comment:
Nit: The `Snapshot` type has a `digest` field, so you could directly request `await Get(Snapshot, RemovePrefix(..))` here to remove one `await`.

Reply:
Concretely, `source_snapshot_stripped = await Get(Snapshot, RemovePrefix(source_digest, source_path))`, followed by `source_snapshot_stripped.digest`

Reply:
Done.